<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="A guide to various plot aesthetic changes that can be done.">

<title>Controlling Plot Aesthetics – dabest</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../images/Favicon-3-outline.svg" rel="icon" type="image/svg+xml">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-80e91010c642a1f9fc1842d203c8c8a0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
<meta property="og:title" content="Controlling Plot Aesthetics – dabest">
<meta property="og:description" content="A guide to various plot aesthetic changes that can be done.">
<meta property="og:image" content="https://acclab.github.io/DABEST-python/tutorials/08-plot_aesthetics_files/figure-html/cell-4-output-1.png">
<meta property="og:site_name" content="dabest">
<meta property="og:image:height" content="551">
<meta property="og:image:width" content="625">
<meta name="twitter:title" content="Controlling Plot Aesthetics – dabest">
<meta name="twitter:description" content="A guide to various plot aesthetic changes that can be done.">
<meta name="twitter:image" content="https://acclab.github.io/DABEST-python/tutorials/08-plot_aesthetics_files/figure-html/cell-4-output-1.png">
<meta name="twitter:image-height" content="551">
<meta name="twitter:image-width" content="625">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">dabest</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../01-getting_started.html" aria-current="page"> 
<span class="menu-text">Get Started</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../tutorials/index.html"> 
<span class="menu-text">Tutorial</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../blog/index.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-help" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Help</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-help">    
        <li>
    <a class="dropdown-item" href="https://github.com/ACCLAB/DABEST-python/issues"><i class="bi bi-bug" role="img">
</i> 
 <span class="dropdown-text">Report an Issue</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/ACCLAB/DABEST-python"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/EstimationStats"> <i class="bi bi-twitter" role="img" aria-label="ACCLAB Twitter">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../tutorials/01-basics.html">Tutorials</a></li><li class="breadcrumb-item"><a href="../tutorials/08-plot_aesthetics.html">Controlling Plot Aesthetics</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../01-getting_started.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Getting Started</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../02-about.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../03-citation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Citing DABEST</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Basics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/02-two_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Two-Group Experiments</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/03-shared_control_and_repeated_measures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Shared Control &amp; Repeated Measures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/04-proportion_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Proportion Plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/05-mini_meta.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Mini-Meta</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/06-delta_delta.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Delta-Delta</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/07-horizontal_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Horizontal Plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/08-plot_aesthetics.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Controlling Plot Aesthetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/09-forest_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Forest Plots: Visualizing Multiple Contrasts</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tutorials/10-whorlmap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Whorlmaps: Visualizing Even More Contrasts</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../API/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">API</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/load.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Loading Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/dabest_object.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dabest object</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/bootstrap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bootstrap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/forest_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Forest plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/plotter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/plot_tools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">plot_tools</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/effsize.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">effsize</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/confint_1group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">confint_1group</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/confint_2group_diff.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">confint_2group_diff</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/delta_objects.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Delta objects</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/misc_tools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">misc_tools</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/effsize_objects.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Effectsize objects</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/precompile.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">precompile</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../API/multi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">multi</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link active" data-scroll-target="#load-libraries">Load libraries</a></li>
  <li><a href="#creating-a-demo-dataset" id="toc-creating-a-demo-dataset" class="nav-link" data-scroll-target="#creating-a-demo-dataset">Creating a demo dataset</a></li>
  <li><a href="#changing-the-graph-colours" id="toc-changing-the-graph-colours" class="nav-link" data-scroll-target="#changing-the-graph-colours">Changing the graph colours</a>
  <ul>
  <li><a href="#color-categories-from-another-variable" id="toc-color-categories-from-another-variable" class="nav-link" data-scroll-target="#color-categories-from-another-variable">Color categories from another variable</a></li>
  <li><a href="#custom-palette" id="toc-custom-palette" class="nav-link" data-scroll-target="#custom-palette">Custom palette</a>
  <ul class="collapse">
  <li><a href="#a-list-of-colors" id="toc-a-list-of-colors" class="nav-link" data-scroll-target="#a-list-of-colors">A list of colors</a></li>
  <li><a href="#seaborn-color-palette" id="toc-seaborn-color-palette" class="nav-link" data-scroll-target="#seaborn-color-palette">Seaborn color palette</a></li>
  <li><a href="#matplotlib-color-mappalette" id="toc-matplotlib-color-mappalette" class="nav-link" data-scroll-target="#matplotlib-color-mappalette">Matplotlib color map/palette</a></li>
  <li><a href="#a-user-defined-dictionary" id="toc-a-user-defined-dictionary" class="nav-link" data-scroll-target="#a-user-defined-dictionary">A user-defined dictionary</a></li>
  <li><a href="#color-palette-changes-also-now-affect-the-effect-size-curve-colors-in-paired-plots" id="toc-color-palette-changes-also-now-affect-the-effect-size-curve-colors-in-paired-plots" class="nav-link" data-scroll-target="#color-palette-changes-also-now-affect-the-effect-size-curve-colors-in-paired-plots">Color palette changes also now affect the effect size curve colors in paired plots</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#color-saturation" id="toc-color-saturation" class="nav-link" data-scroll-target="#color-saturation">Color saturation</a></li>
  <li><a href="#alpha-transparency" id="toc-alpha-transparency" class="nav-link" data-scroll-target="#alpha-transparency">Alpha (transparency)</a></li>
  <li><a href="#marker-size" id="toc-marker-size" class="nav-link" data-scroll-target="#marker-size">Marker size</a></li>
  <li><a href="#axes" id="toc-axes" class="nav-link" data-scroll-target="#axes">Axes</a>
  <ul>
  <li><a href="#lims" id="toc-lims" class="nav-link" data-scroll-target="#lims">Lims</a></li>
  <li><a href="#labels" id="toc-labels" class="nav-link" data-scroll-target="#labels">Labels</a></li>
  <li><a href="#axes-ticks" id="toc-axes-ticks" class="nav-link" data-scroll-target="#axes-ticks">Axes ticks</a></li>
  <li><a href="#add-counts-to-tick-labels" id="toc-add-counts-to-tick-labels" class="nav-link" data-scroll-target="#add-counts-to-tick-labels">Add counts to tick labels</a></li>
  </ul></li>
  <li><a href="#changing-swarm-side" id="toc-changing-swarm-side" class="nav-link" data-scroll-target="#changing-swarm-side">Changing swarm side</a></li>
  <li><a href="#creating-estimation-plots-in-existing-axes" id="toc-creating-estimation-plots-in-existing-axes" class="nav-link" data-scroll-target="#creating-estimation-plots-in-existing-axes">Creating estimation plots in existing axes</a></li>
  <li><a href="#legend" id="toc-legend" class="nav-link" data-scroll-target="#legend">Legend</a></li>
  <li><a href="#hiding-options" id="toc-hiding-options" class="nav-link" data-scroll-target="#hiding-options">Hiding options</a></li>
  <li><a href="#effect-size-error-bar-and-marker" id="toc-effect-size-error-bar-and-marker" class="nav-link" data-scroll-target="#effect-size-error-bar-and-marker">Effect size error bar and marker</a></li>
  <li><a href="#group-summaries" id="toc-group-summaries" class="nav-link" data-scroll-target="#group-summaries">Group summaries</a></li>
  <li><a href="#raw-bars" id="toc-raw-bars" class="nav-link" data-scroll-target="#raw-bars">Raw bars</a></li>
  <li><a href="#contrast-bars" id="toc-contrast-bars" class="nav-link" data-scroll-target="#contrast-bars">Contrast bars</a></li>
  <li><a href="#reference-band" id="toc-reference-band" class="nav-link" data-scroll-target="#reference-band">Reference band</a></li>
  <li><a href="#delta-text" id="toc-delta-text" class="nav-link" data-scroll-target="#delta-text">Delta text</a></li>
  <li><a href="#adding-jitter-to-slopegraph-plots" id="toc-adding-jitter-to-slopegraph-plots" class="nav-link" data-scroll-target="#adding-jitter-to-slopegraph-plots">Adding jitter to slopegraph plots</a></li>
  <li><a href="#gridkey" id="toc-gridkey" class="nav-link" data-scroll-target="#gridkey">Gridkey</a></li>
  <li><a href="#delta-dot" id="toc-delta-dot" class="nav-link" data-scroll-target="#delta-dot">Delta dot</a></li>
  <li><a href="#effect-size-paired-lines" id="toc-effect-size-paired-lines" class="nav-link" data-scroll-target="#effect-size-paired-lines">Effect size paired lines</a></li>
  <li><a href="#baseline-error-curve" id="toc-baseline-error-curve" class="nav-link" data-scroll-target="#baseline-error-curve">Baseline error curve</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/acclab/DABEST-python/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../tutorials/01-basics.html">Tutorials</a></li><li class="breadcrumb-item"><a href="../tutorials/08-plot_aesthetics.html">Controlling Plot Aesthetics</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Controlling Plot Aesthetics</h1>
</div>

<div>
  <div class="description">
    A guide to various plot aesthetic changes that can be done.
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<p>Since <strong>v2024.03.29</strong>, swarmplots are, by default, plotted asymmetrically to the right side. For detailed information, please refer to <a href="#changing-swarm-side">Swarm Side</a>.</p>
<p>Since <strong>v2025.03.27</strong>, further aesthetic changes were added/updated which include:</p>
<ul>
<li><p><a href="#raw-bars">Raw bars</a></p></li>
<li><p><a href="#contrast-bars">Contrast bars</a></p></li>
<li><p><a href="#reference-band">Reference Band</a></p></li>
<li><p><a href="#delta-text">Delta text</a></p></li>
<li><p><a href="#adding-jitter-to-slopegraph-plots">Jitter</a></p></li>
<li><p><a href="#gridkey">Gridkey</a></p></li>
<li><p><a href="#delta-dot">Delta dots</a></p></li>
<li><p><a href="#effect-size-paired-lines">Effect size paired lines</a></p></li>
<li><p><a href="#baseline-error-curve">Baseline error curve</a></p></li>
</ul>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<div id="cell-4" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> dabest</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"We're using DABEST v</span><span class="sc">{}</span><span class="st">"</span>.<span class="bu">format</span>(dabest.__version__))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>Pre-compiling numba functions for DABEST...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Compiling numba functions: 100%|██████████| 11/11 [00:00&lt;00:00, 50.20it/s]</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Numba compilation complete!
We're using DABEST v2025.10.20</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
</div>
</section>
<section id="creating-a-demo-dataset" class="level2">
<h2 class="anchored" data-anchor-id="creating-a-demo-dataset">Creating a demo dataset</h2>
<div id="cell-6" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.stats <span class="im">import</span> norm <span class="co"># Used in generation of populations.</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>np.random.seed(<span class="dv">9999</span>) <span class="co"># Fix the seed to ensure reproducibility of results.</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>Ns <span class="op">=</span> <span class="dv">20</span> <span class="co"># The number of samples taken from each population</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create samples</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>c1 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="dv">3</span>, scale<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>c2 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.5</span>, scale<span class="op">=</span><span class="fl">0.75</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>c3 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.25</span>, scale<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>t1 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.5</span>, scale<span class="op">=</span><span class="fl">0.5</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>t2 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">2.5</span>, scale<span class="op">=</span><span class="fl">0.6</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>t3 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="dv">3</span>, scale<span class="op">=</span><span class="fl">0.75</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>t4 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.5</span>, scale<span class="op">=</span><span class="fl">0.75</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>t5 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.25</span>, scale<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>t6 <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="fl">3.25</span>, scale<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span>Ns)</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a `gender` column for coloring the data.</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>females <span class="op">=</span> np.repeat(<span class="st">'Female'</span>, Ns<span class="op">/</span><span class="dv">2</span>).tolist()</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>males <span class="op">=</span> np.repeat(<span class="st">'Male'</span>, Ns<span class="op">/</span><span class="dv">2</span>).tolist()</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>gender <span class="op">=</span> females <span class="op">+</span> males</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Add an `id` column for paired data plotting.</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>id_col <span class="op">=</span> pd.Series(<span class="bu">range</span>(<span class="dv">1</span>, Ns<span class="op">+</span><span class="dv">1</span>))</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine samples and gender into a DataFrame.</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.DataFrame({<span class="st">'Control 1'</span> : c1,     <span class="st">'Test 1'</span> : t1,</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Control 2'</span> : c2,     <span class="st">'Test 2'</span> : t2,</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Control 3'</span> : c3,     <span class="st">'Test 3'</span> : t3,</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Test 4'</span>    : t4,     <span class="st">'Test 5'</span> : t5, <span class="st">'Test 6'</span> : t6,</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Gender'</span>    : gender, <span class="st">'ID'</span>  : id_col</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>                  })</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>np.random.seed(<span class="dv">9999</span>) <span class="co"># Fix the seed so the results are replicable.</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Create samples</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>N <span class="op">=</span> <span class="dv">20</span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>y <span class="op">=</span> norm.rvs(loc<span class="op">=</span><span class="dv">3</span>, scale<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span>N<span class="op">*</span><span class="dv">4</span>)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>y[N:<span class="dv">2</span><span class="op">*</span>N] <span class="op">=</span> y[N:<span class="dv">2</span><span class="op">*</span>N]<span class="op">+</span><span class="dv">1</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>y[<span class="dv">2</span><span class="op">*</span>N:<span class="dv">3</span><span class="op">*</span>N] <span class="op">=</span> y[<span class="dv">2</span><span class="op">*</span>N:<span class="dv">3</span><span class="op">*</span>N]<span class="op">-</span><span class="fl">0.5</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a `Treatment` column</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>t1 <span class="op">=</span> np.repeat(<span class="st">'Placebo'</span>, N<span class="op">*</span><span class="dv">2</span>).tolist()</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>t2 <span class="op">=</span> np.repeat(<span class="st">'Drug'</span>, N<span class="op">*</span><span class="dv">2</span>).tolist()</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>treatment <span class="op">=</span> t1 <span class="op">+</span> t2 </span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a `Rep` column as the first variable for the 2 replicates of experiments done</span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>rep <span class="op">=</span> []</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(N<span class="op">*</span><span class="dv">2</span>):</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>    rep.append(<span class="st">'Rep1'</span>)</span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>    rep.append(<span class="st">'Rep2'</span>)</span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a `Genotype` column as the second variable</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>wt <span class="op">=</span> np.repeat(<span class="st">'W'</span>, N).tolist()</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>mt <span class="op">=</span> np.repeat(<span class="st">'M'</span>, N).tolist()</span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>wt2 <span class="op">=</span> np.repeat(<span class="st">'W'</span>, N).tolist()</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>mt2 <span class="op">=</span> np.repeat(<span class="st">'M'</span>, N).tolist()</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>genotype <span class="op">=</span> wt <span class="op">+</span> mt <span class="op">+</span> wt2 <span class="op">+</span> mt2</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Add an `id` column for paired data plotting.</span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a><span class="bu">id</span> <span class="op">=</span> <span class="bu">list</span>(<span class="bu">range</span>(<span class="dv">0</span>, N<span class="op">*</span><span class="dv">2</span>))</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>id_col <span class="op">=</span> <span class="bu">id</span> <span class="op">+</span> <span class="bu">id</span> </span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all columns into a DataFrame.</span></span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a>df_delta2 <span class="op">=</span> pd.DataFrame({<span class="st">'ID'</span>        : id_col,</span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a>                  <span class="st">'Rep'</span>      : rep,</span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Genotype'</span>  : genotype, </span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Treatment'</span>: treatment,</span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>                   <span class="st">'Y'</span>         : y</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>                })</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> create_demo_prop_dataset(seed<span class="op">=</span><span class="dv">9999</span>, N<span class="op">=</span><span class="dv">40</span>):</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>    np.random.seed(<span class="dv">9999</span>)  <span class="co"># Fix the seed to ensure reproducibility of results.</span></span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create samples</span></span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> <span class="dv">1</span></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a>    c1 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.2</span>, size<span class="op">=</span>N)</span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>    c2 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.2</span>, size<span class="op">=</span>N)</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>    c3 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.8</span>, size<span class="op">=</span>N)</span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>    t1 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.6</span>, size<span class="op">=</span>N)</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a>    t2 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.2</span>, size<span class="op">=</span>N)</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>    t3 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.3</span>, size<span class="op">=</span>N)</span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>    t4 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.4</span>, size<span class="op">=</span>N)</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a>    t5 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.5</span>, size<span class="op">=</span>N)</span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a>    t6 <span class="op">=</span> np.random.binomial(n, <span class="fl">0.6</span>, size<span class="op">=</span>N)</span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a>    t7 <span class="op">=</span> np.ones(N)</span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>    t8 <span class="op">=</span> np.zeros(N)</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a>    t9 <span class="op">=</span> np.zeros(N)</span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a `gender` column for coloring the data.</span></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a>    females <span class="op">=</span> np.repeat(<span class="st">'Female'</span>, N <span class="op">/</span> <span class="dv">2</span>).tolist()</span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a>    males <span class="op">=</span> np.repeat(<span class="st">'Male'</span>, N <span class="op">/</span> <span class="dv">2</span>).tolist()</span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a>    gender <span class="op">=</span> females <span class="op">+</span> males</span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add an `id` column for paired data plotting.</span></span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a>    id_col <span class="op">=</span> pd.Series(<span class="bu">range</span>(<span class="dv">1</span>, N <span class="op">+</span> <span class="dv">1</span>))</span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Combine samples and gender into a DataFrame.</span></span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> pd.DataFrame({<span class="st">'Control 1'</span>: c1, <span class="st">'Test 1'</span>: t1,</span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a>                       <span class="st">'Control 2'</span>: c2, <span class="st">'Test 2'</span>: t2,</span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a>                       <span class="st">'Control 3'</span>: c3, <span class="st">'Test 3'</span>: t3,</span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a>                       <span class="st">'Test 4'</span>: t4, <span class="st">'Test 5'</span>: t5, <span class="st">'Test 6'</span>: t6,</span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a>                       <span class="st">'Test 7'</span>: t7, <span class="st">'Test 8'</span>: t8, <span class="st">'Test 9'</span>: t9,</span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a>                       <span class="st">'Gender'</span>: gender, <span class="st">'ID'</span>: id_col</span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a>                       })</span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df</span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a>df_prop <span class="op">=</span> create_demo_prop_dataset()</span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a>two_groups_prop_paired <span class="op">=</span> dabest.load(df_prop, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>), proportional<span class="op">=</span><span class="va">True</span>, paired<span class="op">=</span><span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>)</span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a>two_groups_prop <span class="op">=</span> dabest.load(df_prop, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>), proportional<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>))</span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a>multi_2group <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>((<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>),(<span class="st">"Control 2"</span>, <span class="st">"Test 2"</span>)))</span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a>repeated_measures <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>, <span class="st">"Test 2"</span>, <span class="st">"Test 3"</span>),paired<span class="op">=</span><span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>)</span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a>two_groups_paired <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>), paired<span class="op">=</span><span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>)</span>
<span id="cb6-123"><a href="#cb6-123" aria-hidden="true" tabindex="-1"></a>mini_meta_paired <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>((<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>), (<span class="st">"Control 2"</span>, <span class="st">"Test 2"</span>), (<span class="st">"Control 3"</span>, <span class="st">"Test 3"</span>)), mini_meta<span class="op">=</span><span class="va">True</span>, id_col<span class="op">=</span><span class="st">"ID"</span>, paired<span class="op">=</span><span class="st">"baseline"</span>)</span>
<span id="cb6-124"><a href="#cb6-124" aria-hidden="true" tabindex="-1"></a>paired_delta2 <span class="op">=</span> dabest.load(data <span class="op">=</span> df_delta2, </span>
<span id="cb6-125"><a href="#cb6-125" aria-hidden="true" tabindex="-1"></a>                                paired <span class="op">=</span> <span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>,</span>
<span id="cb6-126"><a href="#cb6-126" aria-hidden="true" tabindex="-1"></a>                                x <span class="op">=</span> [<span class="st">"Treatment"</span>, <span class="st">"Rep"</span>], y <span class="op">=</span> <span class="st">"Y"</span>, </span>
<span id="cb6-127"><a href="#cb6-127" aria-hidden="true" tabindex="-1"></a>                                delta2 <span class="op">=</span> <span class="va">True</span>, experiment <span class="op">=</span> <span class="st">"Genotype"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="changing-the-graph-colours" class="level2">
<h2 class="anchored" data-anchor-id="changing-the-graph-colours">Changing the graph colours</h2>
<section id="color-categories-from-another-variable" class="level3">
<h3 class="anchored" data-anchor-id="color-categories-from-another-variable">Color categories from another variable</h3>
<p>Use the parameter <code>color_col</code> to specify which column in the dataframe will be used to create the different colours for your graph.</p>
<div id="cell-8" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(color_col<span class="op">=</span><span class="st">"Gender"</span>)<span class="op">;</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>two_groups_paired.mean_diff.plot(color_col<span class="op">=</span><span class="st">"Gender"</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-4-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-4-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="custom-palette" class="level3">
<h3 class="anchored" data-anchor-id="custom-palette">Custom palette</h3>
<p>The colour palette for the graph can be changed using the parameter <code>custom_palette</code>. Multiple types of color palettes can be used:</p>
<ul>
<li><p>A list of colors (named colors, hex, rgb, etc) e.g.&nbsp;<code>['red', 'blue', 'green']</code></p></li>
<li><p>A seaborn color palette e.g.&nbsp;<code>'Set1'</code></p></li>
<li><p>A matplotlib color map e.g.&nbsp;<code>'viridis'</code></p>
<ul>
<li><code>'paired'</code> is an interesting option for two-group (or multi two-group) comparisons</li>
</ul></li>
<li><p>A dictionary with the keys as the column names and the values as the colors e.g.&nbsp;<code>{'Control 1': 'red', 'Test 1': 'blue', 'Test 2': 'green'}</code></p>
<ul>
<li>Or, a dictionary with the keys as the binary options for proportion plots (barplots and sankey) and the values as the colors e.g.&nbsp;<code>{0: 'red', 1: 'blue'}</code></li>
</ul></li>
</ul>
<section id="a-list-of-colors" class="level4">
<h4 class="anchored" data-anchor-id="a-list-of-colors">A list of colors</h4>
<div id="cell-11" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span>[<span class="st">'red'</span>, <span class="st">'blue'</span>, <span class="st">'green'</span>, <span class="st">'purple'</span>, <span class="st">'orange'</span>, <span class="st">'brown'</span>])<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-5-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="seaborn-color-palette" class="level4">
<h4 class="anchored" data-anchor-id="seaborn-color-palette">Seaborn color palette</h4>
<div id="cell-13" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span>sns.color_palette(<span class="st">"husl"</span>, <span class="dv">6</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-6-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="matplotlib-color-mappalette" class="level4">
<h4 class="anchored" data-anchor-id="matplotlib-color-mappalette">Matplotlib color map/palette</h4>
<div id="cell-15" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span><span class="st">"viridis"</span>)<span class="op">;</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span><span class="st">"Paired"</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-7-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-7-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="a-user-defined-dictionary" class="level4">
<h4 class="anchored" data-anchor-id="a-user-defined-dictionary">A user-defined dictionary</h4>
<p>There are <a href="https://matplotlib.org/users/colors.html">many ways</a> to specify matplotlib colours. Find one example below using accepted colour names, hex strings (commonly used on the web), and RGB tuples.</p>
<div id="cell-17" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>my_color_palette <span class="op">=</span> {<span class="st">"Control 1"</span> : <span class="st">"blue"</span>,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Test 1"</span>    : <span class="st">"purple"</span>,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Control 2"</span> : <span class="st">"#cb4b16"</span>,     <span class="co"># This is a hex string.</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Test 2"</span>    : (<span class="fl">0.</span>, <span class="fl">0.7</span>, <span class="fl">0.2</span>) <span class="co"># This is a RGB tuple.</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                       }</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span>my_color_palette)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-8-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>For proportion plots (barplots and sankey), a color palette dict can also be supplied via <code>{1: first_color, 0, second_color}</code> where first_color and second_color are valid matplotlib colours.</p>
<div id="cell-19" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>two_groups_prop.mean_diff.plot(custom_palette<span class="op">=</span>{<span class="dv">1</span>: <span class="st">"red"</span>, <span class="dv">0</span>: <span class="st">"blue"</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-9-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="cell-20" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>two_groups_prop_paired.mean_diff.plot(custom_palette<span class="op">=</span>{<span class="dv">1</span>: <span class="st">"red"</span>, <span class="dv">0</span>: <span class="st">"blue"</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-10-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="color-palette-changes-also-now-affect-the-effect-size-curve-colors-in-paired-plots" class="level4">
<h4 class="anchored" data-anchor-id="color-palette-changes-also-now-affect-the-effect-size-curve-colors-in-paired-plots">Color palette changes also now affect the effect size curve colors in paired plots</h4>
<p>Note: The first color in the custom palette is used for the control group. As in the example below, if <code>show_baseline_ec</code> is set to <code>False</code>, it wont be represented in the plot.</p>
<div id="cell-22" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(custom_palette<span class="op">=</span>[<span class="st">"red"</span>, <span class="st">"blue"</span>, <span class="st">"green"</span>, <span class="st">"purple"</span>])<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-11-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="color-saturation" class="level2">
<h2 class="anchored" data-anchor-id="color-saturation">Color saturation</h2>
<p>By default, <code>dabest.plot()</code> <a href="https://en.wikipedia.org/wiki/Colorfulness#Saturation">desaturates</a> the colour of the dots in the swarmplot by 50%. This draws attention to the effect size bootstrap curves.</p>
<p>You can alter the default values with the parameters <code>raw_desat</code> and <code>contrast_desat</code>.</p>
<div id="cell-24" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(custom_palette<span class="op">=</span>my_color_palette,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                                raw_desat<span class="op">=</span><span class="fl">0.75</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                                contrast_desat<span class="op">=</span><span class="fl">0.25</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-12-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="alpha-transparency" class="level2">
<h2 class="anchored" data-anchor-id="alpha-transparency">Alpha (transparency)</h2>
<p>It is possible change the transparency of the raw data by using the <code>raw_alpha</code> parameter. This can also be achieved by adding <code>alpha</code> to the relevant rawdata kwargs (<code>barplot_kwargs</code>, or <code>swarmplot_kwargs</code>, or <code>slopegraph_kwargs</code>, or <code>sankey_kwargs</code>)</p>
<div id="cell-26" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(raw_alpha<span class="op">=</span><span class="fl">0.2</span>)<span class="op">;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(swarmplot_kwargs<span class="op">=</span>{<span class="st">'alpha'</span>: <span class="fl">0.2</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-13-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-13-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>It is also possible change the transparency of the effect size curves by using the <code>contrast_alpha</code> parameter. This can also be achieved via adding <code>alpha</code> to the <code>contrast_kwargs</code> parameter.</p>
<div id="cell-28" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(contrast_alpha<span class="op">=</span><span class="fl">0.2</span>)<span class="op">;</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(contrast_kwargs<span class="op">=</span>{<span class="st">'alpha'</span>:<span class="fl">0.2</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-14-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-14-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="marker-size" class="level2">
<h2 class="anchored" data-anchor-id="marker-size">Marker size</h2>
<p>It is possible change the size of the dots used in the rawdata swarmplot, as well as those to indicate the effect sizes, by using the parameters <code>raw_marker_size</code> and <code>contrast_marker_size</code> respectively.</p>
<div id="cell-30" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(raw_marker_size<span class="op">=</span><span class="dv">3</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                                contrast_marker_size<span class="op">=</span><span class="dv">12</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-15-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="axes" class="level2">
<h2 class="anchored" data-anchor-id="axes">Axes</h2>
<section id="lims" class="level3">
<h3 class="anchored" data-anchor-id="lims">Lims</h3>
<p>To change the y-limits for the rawdata axes, and the contrast axes, use the parameters <code>raw_ylim</code> and <code>contrast_ylim</code>.</p>
<div id="cell-33" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(raw_ylim<span class="op">=</span>(<span class="dv">0</span>, <span class="dv">5</span>),</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                                contrast_ylim<span class="op">=</span>(<span class="op">-</span><span class="dv">2</span>, <span class="dv">2</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-16-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>If the effect size is qualitatively inverted (ie. a smaller value is a better outcome), you can simply invert the tuple passed to <code>contrast_ylim</code>.</p>
<div id="cell-35" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(contrast_ylim<span class="op">=</span>(<span class="dv">2</span>, <span class="op">-</span><span class="dv">2</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-17-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>The contrast axes share the same y-limits as those of the delta-delta plot. Thus, the y axis of the delta-delta plot changes as well.</p>
<div id="cell-37" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>paired_delta2.mean_diff.plot(contrast_ylim<span class="op">=</span>(<span class="dv">3</span>, <span class="op">-</span><span class="dv">3</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-18-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>You can also change the y-limit of the delta-delta axes and the regular delta axes via the <code>delta2_ylim</code> parameter.</p>
<div id="cell-39" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>paired_delta2.mean_diff.plot(delta2_ylim<span class="op">=</span>(<span class="dv">3</span>, <span class="op">-</span><span class="dv">3</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-19-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="labels" class="level3">
<h3 class="anchored" data-anchor-id="labels">Labels</h3>
<ul>
<li><code>raw_label</code> - label the raw data y-axis</li>
<li><code>contrast_label</code> - label the contrast y-axis</li>
</ul>
<div id="cell-41" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(raw_label<span class="op">=</span><span class="st">"This is my</span><span class="ch">\n</span><span class="st">rawdata"</span>, </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>                                   contrast_label<span class="op">=</span><span class="st">"The bootstrap</span><span class="ch">\n</span><span class="st">distribtions!"</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>                                )<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-20-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Unique for delta-delta: - <code>delta2_ylim</code> - to label the delta-delta y-axis</p>
<div id="cell-43" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>paired_delta2.mean_diff.plot(delta2_label<span class="op">=</span><span class="st">'delta-delta label'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-21-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="axes-ticks" class="level3">
<h3 class="anchored" data-anchor-id="axes-ticks">Axes ticks</h3>
<p>You can add minor ticks and also change the tick frequency by accessing the axes directly.</p>
<p>Each estimation plot produced by <code>dabest</code> has two axes. The first one contains the rawdata swarmplot while the second one contains the bootstrap effect size differences.</p>
<div id="cell-45" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.ticker <span class="im">as</span> Ticker</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>f <span class="op">=</span> two_groups_unpaired.mean_diff.plot()</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>rawswarm_axes <span class="op">=</span> f.axes[<span class="dv">0</span>]</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>contrast_axes <span class="op">=</span> f.axes[<span class="dv">1</span>]</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>rawswarm_axes.yaxis.set_major_locator(Ticker.MultipleLocator(<span class="dv">1</span>))</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>rawswarm_axes.yaxis.set_minor_locator(Ticker.MultipleLocator(<span class="fl">0.5</span>))</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>contrast_axes.yaxis.set_major_locator(Ticker.MultipleLocator(<span class="fl">0.5</span>))</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>contrast_axes.yaxis.set_minor_locator(Ticker.MultipleLocator(<span class="fl">0.25</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-22-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="add-counts-to-tick-labels" class="level3">
<h3 class="anchored" data-anchor-id="add-counts-to-tick-labels">Add counts to tick labels</h3>
<p>By default, the tick labels include the sample size for each group. This can be switched off via setting <code>show_sample_size=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-47" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(show_sample_size<span class="op">=</span><span class="va">False</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                                )<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-23-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="changing-swarm-side" class="level2">
<h2 class="anchored" data-anchor-id="changing-swarm-side">Changing swarm side</h2>
<p>In <code>dabest</code>, swarmplots are, by default, plotted asymmetrically to the right side. You may change this by using the parameter <code>swarm_side</code>.</p>
<p>There are only three valid values: <code>"right"</code> (default), <code>"left"</code>, <code>"center"</code>.</p>
<div id="cell-49" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(swarm_side<span class="op">=</span><span class="st">"right"</span>)<span class="op">;</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(swarm_side<span class="op">=</span><span class="st">"left"</span>)<span class="op">;</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(swarm_side<span class="op">=</span><span class="st">"center"</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-24-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-24-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-24-output-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="creating-estimation-plots-in-existing-axes" class="level2">
<h2 class="anchored" data-anchor-id="creating-estimation-plots-in-existing-axes">Creating estimation plots in existing axes</h2>
<p><em>Implemented in v0.2.6 by Adam Nekimken</em>.</p>
<p><code>dabest.plot</code> has an <code>ax</code> parameter that accepts Matplotlib <code>Axes</code>. The entire estimation plot will be created in the specified <code>Axes</code>.</p>
<div id="cell-52" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>two_groups_paired_baseline <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>(<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>),</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                                  paired<span class="op">=</span><span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>multi_2group_paired <span class="op">=</span> dabest.load(df,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>                            idx<span class="op">=</span>((<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>),</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>                                 (<span class="st">"Control 2"</span>, <span class="st">"Test 2"</span>)),</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                            paired<span class="op">=</span><span class="st">"baseline"</span>, id_col<span class="op">=</span><span class="st">"ID"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cell-53" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>f, axx <span class="op">=</span> plt.subplots(nrows<span class="op">=</span><span class="dv">2</span>, ncols<span class="op">=</span><span class="dv">2</span>,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                      figsize<span class="op">=</span>(<span class="dv">15</span>, <span class="dv">15</span>),</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                      gridspec_kw<span class="op">=</span>{<span class="st">'wspace'</span>: <span class="fl">0.25</span>} <span class="co"># ensure proper width-wise spacing.</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                     )</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(ax<span class="op">=</span>axx.flat[<span class="dv">0</span>])<span class="op">;</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>two_groups_paired_baseline.mean_diff.plot(ax<span class="op">=</span>axx.flat[<span class="dv">1</span>])<span class="op">;</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(ax<span class="op">=</span>axx.flat[<span class="dv">2</span>])<span class="op">;</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>multi_2group_paired.mean_diff.plot(ax<span class="op">=</span>axx.flat[<span class="dv">3</span>])<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-26-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>In this case, to access the individual rawdata axes, use <code>name_of_axes</code> to manipulate the rawdata axes, and <code>name_of_axes.contrast_axes</code> to gain access to the effect size axes.</p>
<div id="cell-55" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>topleft_axes <span class="op">=</span> axx.flat[<span class="dv">0</span>]</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>topleft_axes.set_ylabel(<span class="st">"New y-axis label for rawdata"</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>topleft_axes.contrast_axes.set_ylabel(<span class="st">"New y-axis label for effect size"</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>f</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-27-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="legend" class="level2">
<h2 class="anchored" data-anchor-id="legend">Legend</h2>
<p>For plots with a <code>color_col</code> specified, a legend will be created. Utilise the <code>legend_kwargs</code> parameter to adjust the legend.</p>
<div id="cell-57" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(color_col<span class="op">=</span><span class="st">"Gender"</span>, </span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>                            legend_kwargs<span class="op">=</span>{<span class="st">'bbox_to_anchor'</span>: [<span class="dv">0</span>, <span class="dv">1</span>], <span class="st">'fontsize'</span>:<span class="dv">8</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-28-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="hiding-options" class="level2">
<h2 class="anchored" data-anchor-id="hiding-options">Hiding options</h2>
<p>For mini-meta plots, it is possible to hide the weighted average plot by setting the parameter <code>show_mini_meta=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-59" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>mini_meta_paired.mean_diff.plot(show_mini_meta<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-29-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Similarly, you can hide the delta-delta effect size by setting <code>show_delta2=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-61" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>paired_delta2.mean_diff.plot(show_delta2<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-30-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-size-error-bar-and-marker" class="level2">
<h2 class="anchored" data-anchor-id="effect-size-error-bar-and-marker">Effect size error bar and marker</h2>
<p>Modifying the effect size marker can be done via <code>contrast_marker_kwargs</code>. This parameter accepts a dictionary of keyword arguments.</p>
<p>The available options are:</p>
<ul>
<li><code>'marker'</code> - type of the marker</li>
<li><code>'markersize'</code> - size of the marker</li>
<li><code>'color'</code> - color of the marker</li>
<li><code>'alpha'</code> - alpha of the marker (transparency)</li>
<li><code>'zorder'</code> - zorder of the marker (the layering relative to other plot elements)</li>
</ul>
<p><strong>Note: markersize can also be modified directly via the <code>contrast_marker_size</code> argument</strong></p>
<div id="cell-63" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(contrast_marker_kwargs<span class="op">=</span>{<span class="st">"marker"</span>: <span class="st">"x"</span>, <span class="st">'markersize'</span>: <span class="dv">15</span>, <span class="st">'color'</span>: <span class="st">'green'</span>, <span class="st">'alpha'</span>:<span class="fl">0.8</span>, <span class="st">'zorder'</span>: <span class="dv">5</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-31-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Modifying the appearance of the effect size error bar can be done via the <code>contrast_errorbar_kwargs</code> parameter. This parameter accepts a dictionary of keyword arguments.</p>
<p>The relevant inputs to <code>contrast_errorbar_kwargs</code> are:</p>
<ul>
<li><code>'lw'</code> - width of the error bar</li>
<li><code>'linestyle'</code> - line style of the error bar</li>
<li><code>'color'</code> - color of the error bar</li>
<li><code>'zorder'</code> - zorder of the error bar (the layering relative to other plot elements)</li>
<li><code>'alpha'</code> - alpha of the error bar (transparency)</li>
</ul>
<div id="cell-65" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(contrast_errorbar_kwargs<span class="op">=</span>{<span class="st">'lw'</span>: <span class="dv">4</span>, <span class="st">'color'</span>: <span class="st">'green'</span>, <span class="st">'alpha'</span>:<span class="fl">0.5</span>, <span class="st">'zorder'</span>: <span class="dv">2</span>, <span class="st">'linestyle'</span>: <span class="st">':'</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-32-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="group-summaries" class="level2">
<h2 class="anchored" data-anchor-id="group-summaries">Group summaries</h2>
<p>Group summaries represent the summary statistics of the sample and are included by default.</p>
<p>In swarmplots and proportion plots, these are represented by gapped lines.</p>
<p>In slopegraphs, these are represented by a solid line connecting the group mean/median with error bars.</p>
<p>The type of group summary can be specified via <code>group_summaries</code> in the <code>.plot()</code> method and must be one of these: <code>'median_quartiles'</code>, <code>'mean_sd'</code>, <code>None</code>.</p>
<p>By default, the group summary is set to <code>'mean_sd'</code>.</p>
<div id="cell-68" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(group_summaries<span class="op">=</span><span class="st">"mean_sd"</span>)<span class="op">;</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(group_summaries<span class="op">=</span><span class="st">"median_quartiles"</span>)<span class="op">;</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(group_summaries<span class="op">=</span><span class="va">None</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-33-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-33-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-33-output-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>For slopegraphs, the group summary is represented by a solid line connecting the group mean/median with error bars.</p>
<div id="cell-70" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(group_summaries<span class="op">=</span><span class="st">"mean_sd"</span>)<span class="op">;</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(group_summaries<span class="op">=</span><span class="st">"median_quartiles"</span>)<span class="op">;</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(group_summaries<span class="op">=</span><span class="va">None</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-34-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-34-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-34-output-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p><strong>Group summaries</strong> have an associated kwargs <code>group_summaries_kwargs</code></p>
<p>The relevant inputs to <code>group_summaries_kwargs</code> are:</p>
<ul>
<li><code>'zorder'</code> - zorder of the gapped lines (the layering relative to other plot elements)</li>
<li><code>'lw'</code> - linewidth of the gapped lines</li>
<li><code>'alpha'</code> - alpha of the gapped lines (transparency)</li>
<li><code>'gap_width_percent'</code> - gap size (for gapped lines only)</li>
<li><code>'offset'</code> - location adjustment of the gapped lines (x-axis; for gapped lines only)</li>
<li><code>'color'</code> - the shared color of the gapped lines</li>
</ul>
<div id="cell-72" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>two_groups_unpaired.mean_diff.plot(group_summaries_kwargs<span class="op">=</span>{<span class="st">'gap_width_percent'</span>: <span class="dv">3</span>, <span class="st">'alpha'</span>: <span class="fl">0.5</span>, <span class="st">'lw'</span>: <span class="dv">4</span>, <span class="st">'offset'</span>: <span class="fl">0.6</span>, <span class="st">'color'</span>:<span class="st">'red'</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-35-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="cell-73" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(group_summaries_kwargs<span class="op">=</span>{<span class="st">'gap_width_percent'</span>: <span class="dv">3</span>, <span class="st">'alpha'</span>: <span class="fl">0.5</span>, <span class="st">'lw'</span>: <span class="dv">4</span>, <span class="st">'offset'</span>: <span class="fl">0.6</span>, <span class="st">'color'</span>:<span class="st">'red'</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-36-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="raw-bars" class="level2">
<h2 class="anchored" data-anchor-id="raw-bars">Raw bars</h2>
<p><strong>Raw bars</strong> are included in swarmplots by default. It can be turned off by setting <code>raw_bars=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-76" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(raw_bars<span class="op">=</span><span class="va">True</span>, contrast_bars<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-37-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Raw bar kwargs can be utilised via <code>raw_bars_kwargs</code> in the <code>.plot()</code> method.</p>
<p>Pass any keyword arguments accepted by matplotlib.patches.Rectangle here, as a string.</p>
<div id="cell-78" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(raw_bars<span class="op">=</span><span class="va">True</span>, contrast_bars<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                            raw_bars_kwargs<span class="op">=</span>{<span class="st">'color'</span>: <span class="st">"red"</span>, <span class="st">'alpha'</span>: <span class="fl">0.2</span>}, </span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                            )<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-38-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="contrast-bars" class="level2">
<h2 class="anchored" data-anchor-id="contrast-bars">Contrast bars</h2>
<p><strong>Contrast bars</strong> are included in all plots by default. It can be turned off by setting <code>contrast_bars=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-80" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(contrast_bars<span class="op">=</span><span class="va">True</span>, raw_bars<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-39-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Contrast bar kwargs can be utilised via <code>contrast_bars_kwargs</code> in the <code>.plot()</code> method.</p>
<p>Pass any keyword arguments accepted by matplotlib.patches.Rectangle here, as a string.</p>
<div id="cell-82" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(contrast_bars<span class="op">=</span><span class="va">True</span>, raw_bars<span class="op">=</span><span class="va">False</span>, </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                            contrast_bars_kwargs<span class="op">=</span>{<span class="st">'color'</span>: <span class="st">"red"</span>, <span class="st">'alpha'</span>: <span class="fl">0.1</span>}</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                            )<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-40-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="reference-band" class="level2">
<h2 class="anchored" data-anchor-id="reference-band">Reference band</h2>
<p>A <strong>reference band</strong> can be added for each relevant contrast object as desired via supplying a list to the argument <code>reference_band</code> in the <code>.plot()</code> method.</p>
<div id="cell-84" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(reference_band<span class="op">=</span>[<span class="dv">0</span>, <span class="dv">1</span>], contrast_bars<span class="op">=</span><span class="va">False</span>, raw_bars<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-41-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Reference band kwargs can be utilised via <code>reference_band_kwargs</code> in the <code>.plot()</code> method.</p>
<p>The relevant inputs to <code>reference_band_kwargs</code> are:</p>
<ul>
<li><code>'span_ax'</code> - Whether the reference band(s) should span the entire x-axis or start from the relevant effect size curve</li>
<li><code>'color'</code> - Color of the reference band(s). If color is not specified, the color of the effect size curve will be used.</li>
<li><code>'alpha'</code> - Alpha of the reference band(s) (transparency)</li>
<li><code>'zorder'</code> - Zorder of the reference band(s) (the layering relative to other plot elements)</li>
</ul>
<div id="cell-86" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(reference_band<span class="op">=</span>[<span class="dv">0</span>,<span class="dv">1</span>], contrast_bars<span class="op">=</span><span class="va">False</span>, raw_bars<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                            reference_band_kwargs<span class="op">=</span>{<span class="st">"alpha"</span>: <span class="fl">0.2</span>, <span class="st">"color"</span>: <span class="st">'black'</span>, <span class="st">'span_ax'</span>: <span class="va">True</span>}</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                            )<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-42-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="delta-text" class="level2">
<h2 class="anchored" data-anchor-id="delta-text">Delta text</h2>
<p><strong>Delta text</strong> is included in all plots by default. It can be turned off by setting <code>delta_text=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-88" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(delta_text<span class="op">=</span><span class="va">True</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-43-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Delta text kwargs can be utilised via <code>delta_text_kwargs</code> in the <code>.plot()</code> method.</p>
<p>The relevant inputs to <code>delta_text_kwargs</code> are:</p>
<ul>
<li><code>'color'</code> - Color. If color is not specified, the color of the effect size curve will be used.</li>
<li><code>'alpha'</code>- Alpha (transparency)</li>
<li><code>'fontsize'</code> - Font size</li>
<li><code>'ha'</code> - Horizontal alignment</li>
<li><code>'va'</code> - Vertical alignment</li>
<li><code>'rotation'</code> - Text rotation</li>
<li><code>'x_coordinates'</code> - Specify the x-coordinates of the text</li>
<li><code>'y_coordinates'</code> - Specify the y-coordinates of the text</li>
<li><code>'offset'</code> - Am x-axis coordinate adjuster for minor movement of all text</li>
</ul>
<p>Otherwise, pass any keyword arguments accepted by matplotlib.text.Text, as a string.</p>
<div id="cell-90" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(delta_text<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>                            delta_text_kwargs<span class="op">=</span>{<span class="st">"color"</span>:<span class="st">"red"</span>, <span class="st">"rotation"</span>:<span class="dv">45</span>, <span class="st">"va"</span>:<span class="st">"bottom"</span>, <span class="st">"alpha"</span>:<span class="fl">0.7</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-44-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p><code>'x_coordinates'</code> and/or <code>'y_coordinates'</code> if you would like to specify the text locations manually.</p>
<div id="cell-92" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(delta_text<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>                            delta_text_kwargs<span class="op">=</span>{<span class="st">"x_coordinates"</span>:(<span class="fl">0.5</span>, <span class="fl">2.75</span>), </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">"y_coordinates"</span>:(<span class="fl">0.5</span>, <span class="op">-</span><span class="fl">1.7</span>)})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-45-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p><code>'offset'</code> to adjust the x location of all the texts (positive moves right, negative left).</p>
<div id="cell-94" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(delta_text<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>                            delta_text_kwargs<span class="op">=</span>{<span class="st">"offset"</span>:<span class="fl">0.1</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-46-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="adding-jitter-to-slopegraph-plots" class="level2">
<h2 class="anchored" data-anchor-id="adding-jitter-to-slopegraph-plots">Adding jitter to slopegraph plots</h2>
<p>For paired plots, you can add jitter to the slopegraph by adding a value for <code>jitter</code> in the <code>slopegraph_kwargs</code> parameter.</p>
<p>This can be useful for specific paired plots when there are many overlapping points.</p>
<p>Currently, jitter is only available for slopegraphs and only in the x-direction (vertical plots) or y-direction (horizontal plots).</p>
<div id="cell-96" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Jitter tests</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>np.random.seed(<span class="dv">9999</span>) <span class="co"># Fix the seed to ensure reproducibility of results.</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>Ns <span class="op">=</span> <span class="dv">20</span> <span class="co"># The number of samples taken from each population</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create samples</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>c1 <span class="op">=</span> [<span class="fl">0.5</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="fl">1.5</span>]<span class="op">*</span>Ns</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>c2 <span class="op">=</span> [<span class="dv">2</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="dv">1</span>]<span class="op">*</span>Ns</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>t1 <span class="op">=</span> [<span class="dv">1</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="dv">2</span>]<span class="op">*</span>Ns</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>t2 <span class="op">=</span> [<span class="fl">1.5</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="fl">2.5</span>]<span class="op">*</span>Ns</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>t3 <span class="op">=</span> [<span class="dv">2</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="dv">1</span>]<span class="op">*</span>Ns</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>t4 <span class="op">=</span> [<span class="dv">1</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="dv">2</span>]<span class="op">*</span>Ns</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>t5 <span class="op">=</span> [<span class="fl">1.5</span>]<span class="op">*</span>Ns <span class="op">+</span> [<span class="fl">2.5</span>]<span class="op">*</span>Ns</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>id_col <span class="op">=</span> pd.Series(<span class="bu">range</span>(<span class="dv">1</span>, <span class="dv">2</span><span class="op">*</span>Ns<span class="op">+</span><span class="dv">1</span>))</span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>df_jittertest<span class="op">=</span> pd.DataFrame({<span class="st">'Control 1'</span> : c1,     <span class="st">'Test 1'</span> : t1,</span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>                 <span class="st">'Control 2'</span> : c2,     <span class="st">'Test 2'</span> : t2, <span class="st">'Test 3'</span> : t3,</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>                    <span class="st">'Test 4'</span>    : t4,     <span class="st">'Test 5'</span> : t5, <span class="st">'ID'</span>  : id_col})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>For the example below, there are many overlapping points for the paired plot, which makes it look like only one sample.</p>
<div id="cell-98" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>multi_2group <span class="op">=</span> dabest.load(df_jittertest, idx<span class="op">=</span>((<span class="st">"Control 1"</span>, <span class="st">"Test 1"</span>),(<span class="st">"Control 2"</span>, <span class="st">"Test 2"</span>, <span class="st">"Test 3"</span>, <span class="st">"Test 4"</span>, <span class="st">"Test 5"</span>)), paired<span class="op">=</span><span class="st">'baseline'</span>, id_col<span class="op">=</span><span class="st">'ID'</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(horizontal<span class="op">=</span><span class="va">False</span>, group_summaries<span class="op">=</span><span class="va">None</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-48-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Adding jitter can help to visualize the data better.</p>
<div id="cell-100" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(horizontal<span class="op">=</span><span class="va">False</span>, slopegraph_kwargs<span class="op">=</span>{<span class="st">'jitter'</span>: <span class="dv">1</span>}, group_summaries<span class="op">=</span><span class="va">None</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-49-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="gridkey" class="level2">
<h2 class="anchored" data-anchor-id="gridkey">Gridkey</h2>
<p>You can utilise a gridkey table format for representing the index groupings. This can be reached via <code>gridkey</code> in the <code>.plot()</code> method.</p>
<p>You can either use <code>gridkey='auto'</code> to automatically generate the gridkey, or pass a list of indexes to represent the groupings (e.g., <code>gridkey=['Control', 'Test']</code>).</p>
<div id="cell-102" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>paired_delta2.mean_diff.plot(gridkey<span class="op">=</span><span class="st">'auto'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-50-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Gridkey kwargs can be utilised via <code>gridkey_kwargs</code> in the <code>.plot()</code> method.</p>
<p>The relevant inputs to <code>gridkey_kwargs</code> are:</p>
<ul>
<li><code>'show_es'</code> - Whether to show the effect size in the gridkey</li>
<li><code>'show_Ns'</code> - Whether to show the sample sizes in the gridkey</li>
<li><code>'merge_pairs'</code> - Whether to merge the pairs in the gridkey (paired data only)</li>
<li><code>'delimiters'</code> - Delimiters to use for the autoparser. E.g., [‘;’, ‘&gt;’, ’_’]</li>
<li><code>'marker'</code> - Marker to use for filling the gridkey</li>
<li><code>'fontsize'</code> - Font size of the gridkey text</li>
<li><code>'labels_fontsize'</code> - Font size of the labels in the gridkey</li>
</ul>
<div id="cell-104" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>multi_2group <span class="op">=</span> dabest.load(df, idx<span class="op">=</span>((<span class="st">"Control 1"</span>, <span class="st">"Control 2"</span>), (<span class="st">"Test 1"</span>, <span class="st">"Test 2"</span>)),paired<span class="op">=</span><span class="st">'baseline'</span>, id_col<span class="op">=</span><span class="st">'ID'</span>)</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>multi_2group.mean_diff.plot(gridkey<span class="op">=</span>[<span class="st">'Control'</span>, <span class="st">'Test'</span>], </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>                            gridkey_kwargs<span class="op">=</span>{<span class="st">'merge_pairs'</span>: <span class="va">True</span>, <span class="st">'show_es'</span>: <span class="va">False</span>, <span class="st">'show_Ns'</span>: <span class="va">False</span>, <span class="st">'marker'</span>: <span class="st">'√'</span>,</span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'fontsize'</span>: <span class="dv">8</span>, <span class="st">'labels_fontsize'</span>: <span class="dv">12</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-51-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="delta-dot" class="level2">
<h2 class="anchored" data-anchor-id="delta-dot">Delta dot</h2>
<p>By default, delta dots are included in paired experiment plots (excluding proportion plots).</p>
<p>This feature can be turned off by setting <code>delta_dot=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-106" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>multi_2group_paired.mean_diff.plot(delta_dot<span class="op">=</span><span class="va">False</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-52-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Delta dot kwargs can be utilised via <code>delta_dot_kwargs</code> in the <code>.plot()</code> method.</p>
<p>The relevant inputs to <code>delta_dot_kwargs</code> are:</p>
<ul>
<li><code>'color'</code> - Specify the color of the delta dots. If color is not specified, the color of the effect size curve will be used.</li>
<li><code>'marker'</code> - Marker of the dots. The default are triangles (‘^’)</li>
<li><code>'alpha'</code> - Alpha (Transparency)</li>
<li><code>'zorder'</code> - Zorder (the layering relative to other plot elements)</li>
<li><code>'size'</code> - Marker size</li>
<li><code>'side'</code> - Which side to plot the delta dots. The options are <code>'left'</code>, <code>'right'</code>, or <code>'center'</code>. This functions like the <code>swarm_side</code> parameter.</li>
</ul>
<div id="cell-108" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>multi_2group_paired.mean_diff.plot(delta_dot_kwargs<span class="op">=</span>{<span class="st">"color"</span>:<span class="st">'red'</span>, <span class="st">"alpha"</span>:<span class="fl">0.1</span>, <span class="st">'zorder'</span>: <span class="dv">2</span>, <span class="st">'size'</span>: <span class="dv">5</span>, <span class="st">'side'</span>: <span class="st">'center'</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-53-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-size-paired-lines" class="level2">
<h2 class="anchored" data-anchor-id="effect-size-paired-lines">Effect size paired lines</h2>
<p>By default, effect size paired lines are included in paired experiment plots (excluding proportion plots).</p>
<p>This feature can be turned off by setting <code>contrast_paired_lines=False</code> in the <code>.plot()</code> method.</p>
<div id="cell-110" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(contrast_paired_lines<span class="op">=</span><span class="va">True</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-54-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<p>Effect size line kwargs can be utilised via <code>contrast_paired_lines_kwargs</code> in the <code>.plot()</code> method.</p>
<p>By default, the following keywords are passed:</p>
<ul>
<li><code>'linestyle'</code> - Linestyle</li>
<li><code>'linewidth'</code> - Linewidth</li>
<li><code>'zorder'</code> - Zorder (the layering relative to other plot elements)</li>
<li><code>'color'</code> - Color. Default is ‘dimgray’</li>
<li><code>'alpha'</code> - Alpha (transparency)</li>
</ul>
<div id="cell-112" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(contrast_paired_lines<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>                                 contrast_paired_lines_kwargs<span class="op">=</span>{<span class="st">"color"</span>: <span class="st">"red"</span>, <span class="st">"alpha"</span>: <span class="fl">0.5</span>, <span class="st">"linestyle"</span>: <span class="st">"-"</span>})<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-55-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="baseline-error-curve" class="level2">
<h2 class="anchored" data-anchor-id="baseline-error-curve">Baseline error curve</h2>
<p>In DABEST <strong>v2025.03.27</strong>, we introduce a new aspect to the contrast axes: the baseline dot and error curve. While the baseline dot is always present, the error curve can be turned on by setting <code>show_baseline_ec=True</code> in the <code>.plot()</code> method.</p>
<div id="cell-114" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>repeated_measures.mean_diff.plot(show_baseline_ec<span class="op">=</span><span class="va">True</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="08-plot_aesthetics_files/figure-html/cell-56-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/acclab\.github\.io\/DABEST-python");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/acclab/DABEST-python/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>